File: Algorithms/Python/BinarySearch.py
def binary_search(arr, to_find, low_pointer, high_pointer):
    if high_pointer >= low_pointer:
        mid = (low_pointer + high_pointer) // 2
        if arr[mid] == to_find:
            return mid
        if arr[mid] > to_find:
            return binary_search(arr, to_find, low_pointer, mid - 1)
        return binary_search(arr, to_find, mid + 1, high_pointer)
    return -1


test_arr = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
find_val = 3
result = binary_search(test_arr, find_val, 0, len(test_arr) - 1)
if result == -1:
    print("The value (" + str(find_val) + ") was not found.")
else:
    print("The value (" + str(find_val) + ") was found!")

File: Algorithms/JAVA/src/implementing_algorithms/BinarySearch.java
package implementing_algorithms;

public class BinarySearch {

	public int binarySearch(int arr[], int toFind, int lowP, int highP) {
		if (highP >= lowP) {
			int mid = (lowP + highP) / 2;
			if (arr[mid] == toFind)
				return mid;

			if (arr[mid] > toFind)
				return binarySearch(arr, toFind, lowP, mid - 1);
			else
				return binarySearch(arr, toFind, mid + 1, highP);
		}
		return -1; // Not found
	}

	public static void main(String[] args) {
		BinarySearch myBS = new BinarySearch();
		int testArr[] = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 };
		int findVal = 3;
		int result = myBS.binarySearch(testArr, findVal, 0, testArr.length - 1);
		if (result == -1)
			System.out.println("The value (" + findVal + ") was not found.");
		else
			System.out.println("The value (" + findVal + ") was found!");
	}
}